OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs

```diff
@@ -96,23 +96,31 @@ public override decimal GetToBePaidAmount(decimal ExpectedAmount, decimal Transf
         public override async Task<PostReqCNResponse> PostReqControlNumberAsync(string OfficerCode, string PaymentId, string PhoneNumber, decimal ExpectedAmount, List<PaymentDetail> products, string controlNumber = null, bool acknowledge = false, bool error = false)
         {
             GepgUtility gepg = new GepgUtility(_hostingEnvironment,config);
-            var bill = gepg.CreateBill(Configuration, OfficerCode, PhoneNumber, PaymentId, Math.Round(ExpectedAmount,2), InsureeProducts);
-            var signature = gepg.GenerateSignature(bill);
 
-            var signedMesg = gepg.FinaliseSignedMsg(signature);
-            var billAck = await gepg.SendHttpRequest("/api/bill/sigqrequest", signedMesg, gepg.GetAccountCodeByProductCode(InsureeProducts.FirstOrDefault().ProductCode), "default.sp.in");
+            ExpectedAmount = Math.Round(ExpectedAmount, 2);
+            //send request only when we have amount > 0
+            if (ExpectedAmount > 0)
+            {
+                var bill = gepg.CreateBill(Configuration, OfficerCode, PhoneNumber, PaymentId, ExpectedAmount, InsureeProducts);
+                var signature = gepg.GenerateSignature(bill);
 
-            string mydocpath = System.IO.Path.Combine(env.WebRootPath, "controlNumberAck");
-            string namepart = new Random().Next(100000, 999999).ToString();
+                var signedMesg = gepg.FinaliseSignedMsg(signature);
+                var billAck = await gepg.SendHttpRequest("/api/bill/sigqrequest", signedMesg, gepg.GetAccountCodeByProductCode(InsureeProducts.FirstOrDefault().ProductCode), "default.sp.in");
 
-            string reconc = JsonConvert.SerializeObject(billAck);
-            string sentbill = JsonConvert.SerializeObject(bill);
+                string reconc = JsonConvert.SerializeObject(billAck);
+                string sentbill = JsonConvert.SerializeObject(bill);
 
-            var content = sentbill + "********************" + reconc;
-            var gepgFile = new GepgFoldersCreating(PaymentId, "CN_Request", content, env);
-            gepgFile.putToTargetFolderPayment();
+                var content = sentbill + "********************" + reconc;
+                var gepgFile = new GepgFoldersCreating(PaymentId, "CN_Request", content, env);
```

Comment:
Saving data in files is always same code on 2 lines. Please create a static method `log` in GePGFileLogger class (change class name) and call it with same parameters. Then you will have only one line of code and no variables.

Reply:
ok